lntest/itest/lnd_test.go

```diff
@@ -1762,6 +1762,13 @@ out:
 		select {
 		case graphUpdate := <-subscription.updateChan:
 			for _, update := range graphUpdate.ChannelUpdates {
+				if len(expUpdates) == 0 {
```

Comment:
Does this assertion need to be in the inner loop? Appears it can reside right when we get a channel read (and check the length of `graphUpdate.ChannelUpdates`, as in that case we got an update and we may not be expecting one at all.

Reply:
I wrote it here so that we could get more debug info from the failure.